### PR TITLE
remove direct refs to homeassistant.helpers when HA >= 2024.6

### LINKED ISSUE #229: Error: async_get() takes 1 positional argument but 2 were given
I'm getting this new error for a `zha_toolkit.attr_read` after updating from `2024.5.5` to `2024.6.0`.

### PR DESCRIPTION
**Pull request by DerekF35**
Closes #229
Apprach taken from https://github.com/custom-components/ble_monitor/commit/a95ae14a2146ff6db009ef428e8aeb5733c47238

### CODE REVIEW

**Review by wbyoung — APPROVED** [review]
This looks like a reasonable solution to me. However, like you, I don't do much coding with HA core, so my review is more like a sanity check.

**Comment by CordoWEB** [comment]
All is working fine with this patch by i do not know if it is the best practices :)
Tested with HA 2024.6.2.

**Review by CordoWEB — APPROVED** [review]
All is working fine with this patch but i do not know if it is the best practices :)
Tested on HA 2024.6.2.

File: custom_components/zha_toolkit/utils.py [hunk 1]
```diff
@@ -34,6 +34,9 @@
     # pylint: disable=ungrouped-imports
     from homeassistant.helpers.json import save_json
 
+if parse_version(HA_VERSION) >= parse_version("2024.6"):
+    from homeassistant.helpers import device_registry, entity_registry
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:38]: Suggestion based on pattern in Home-Assistant Core: rename the imported registries

```suggestion
    from homeassistant.helpers import device_registry as dr, entity_registry as er
```

File: custom_components/zha_toolkit/utils.py [hunk 2]
```diff
@@ -277,7 +280,7 @@ async def get_ieee(app, listener, ref):
             return device.ieee
 
         # Todo: check if NWK address
-        entity_registry = (
+        ent_registry = (
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:283]:
```suggestion
        entity_registry = (
```

File: custom_components/zha_toolkit/utils.py [hunk 3]
```diff
@@ -286,9 +289,11 @@ async def get_ieee(app, listener, ref):
             else get_hass(listener).helpers.entity_registry.async_get(
                 get_hass(listener)
             )
+            if not is_ha_ge("2024.6")
+            else entity_registry.async_get(get_hass(listener))
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:293]:
```suggestion
            else er.async_get(get_hass(listener))
```

File: custom_components/zha_toolkit/utils.py [hunk 4]
```diff
@@ -286,9 +289,11 @@ async def get_ieee(app, listener, ref):
...
         )
 
-        device_registry = (
+        dev_registry = (
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:296]:
```suggestion
        device_registry = (
```

File: custom_components/zha_toolkit/utils.py [hunk 5]
```diff
@@ -297,12 +302,14 @@ async def get_ieee(app, listener, ref):
             else get_hass(listener).helpers.device_registry.async_get(
                 get_hass(listener)
             )
+            if not is_ha_ge("2024.6")
+            else device_registry.async_get(get_hass(listener))
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:306]:
```suggestion
            else dr.async_get(get_hass(listener))
```

File: custom_components/zha_toolkit/utils.py [hunk 6]
```diff
@@ -297,12 +302,14 @@ async def get_ieee(app, listener, ref):
...
         )
-        registry_device = device_registry.async_get(ref)
+        registry_device = dev_registry.async_get(ref)
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:308]:
```suggestion
        registry_device = device_registry.async_get(ref)
```

File: custom_components/zha_toolkit/utils.py [hunk 7]
```diff
@@ -297,12 +302,14 @@ async def get_ieee(app, listener, ref):
...
 
         if registry_device is None:
             # LOGGER.debug("registry %s",entity_registry)
-            registry_entity = entity_registry.async_get(ref)
+            registry_entity = ent_registry.async_get(ref)
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:312]:
```suggestion
            registry_entity = entity_registry.async_get(ref)
```

File: custom_components/zha_toolkit/utils.py [hunk 8]
```diff
@@ -311,7 +318,7 @@ async def get_ieee(app, listener, ref):
                 return None
 
             LOGGER.debug("Found registry_entity %r", registry_entity)
-            registry_device = device_registry.async_get(
+            registry_device = dev_registry.async_get(
```
Comment (Teagan42) [inline custom_components/zha_toolkit/utils.py:321]:
```suggestion
            registry_device = device_registry.async_get(
```

**Review by Teagan42 — APPROVED** [review]
Non-blocking changes, simply suggestion based on the patterns in HA core:
https://github.com/home-assistant/core/blob/2ad5b1c3a6140a49d1113e86e46b68165cf26884/homeassistant/helpers/device.py#L5
https://github.com/home-assistant/core/blob/2ad5b1c3a6140a49d1113e86e46b68165cf26884/homeassistant/auth/permissions/models.py#L10
https://github.com/home-assistant/core/blob/2ad5b1c3a6140a49d1113e86e46b68165cf26884/homeassistant/components/nest/__init__.py#L49

**Comment by DerekF35** [comment]
@Teagan42 Ah yes, that is much simpler and cleaner.  I've made the code changes locally and everything is still in working order.  I've repushed after updating with your suggestions.

**Comment by clarkewing** [comment]
@mdeweerd any timeline on a merge?

**Comment by wbyoung** [comment]
I finally got tired of waiting for this to be merged & released and patched my install. I can confirm that it's working well for me as well.

**Comment by macecchi** [comment]
I can confirm this fix works here too. I'm on HA 2024.6.4.

**Comment by owpac** [comment]
Hello, same problem for me. The fix work locally too. Would it be better if this was merged tho 🙏

**Comment by mdeweerd** [comment]
Hello everybody

Sorry for my absence on github - I've been very busy and I had to drop some activities.

I am looking into the PRs and integrate them in the main branch.

**Comment by CordoWEB** [comment]
Tu es excusé :)
Si tu as besoin d'aide je peux potentiellement le faire. Webmaster et développeur C# à mes heures.

**Pull request merged**